javascript/linked-list/linkedList.js

```diff
@@ -45,23 +45,125 @@ class LinkedList {
       current = current.next;
     }
     return false;
-  };
+  }
 
   // toString
   // "{ a } -> { b } -> { c } -> NULL"
   toString() {
-    if (!this.head) return "NULL";
+    if (!this.head) return 'NULL';
     let current = this.head;
-    let returnStr = "";
+    let returnStr = '';
     while (current) {
       // look at the node.value "{ a } -> "
       returnStr += `{ ${current.value} } -> `;
       current = current.next;
     }
     // once we exit the loop we need to tack NULL on to the end
-    returnStr += "NULL";
+    returnStr += 'NULL';
     return returnStr;
   }
+
+
+
+
+  // append
+  // arguments: new value
+  // adds a new node with the given value to the end of the list
+  // - create a function that takes in a new value
+  // - make a new node with the value argument
+  // - traverse to the end of linke list
+  // - end node.next will be the new node
+
+  append(newValue) {
+    const newNode = new Node(newValue);
+    let current = this.head;
+    while(current.next) {
+      current = current.next;
+    }
+    current.next = newNode;
+  }
+
+
+  // const addToEnd = (newValue) => {
+  //   var newEnd = new Node(newValue);
+  //   if (head === null) {
+  //     head = new Node(newValue);
+  //     return;
+  //   }
+  //   newEnd.next = null;
+  //   var last = head;
+  //   while (last.next != null)
+  //     last = last.next;
+  //   last.next = newEnd;
+  //   return;
+  // }
+
+  // const addToEnd = (newValue) => {
+  //   let newEnd = new Node(newValue);
+  //   var current = this.head;
+  //   while (current) {
+  //     current = current.next
+  //   }
+  //    current.next = newEnd;
+  // }
+
+
+
+
+  // insert before
+  // arguments: value, new value
+  // adds a new node with the given new value immediately before the first node that has the value specified
+  // -assign node to temp var
+  //- if heads not value start looping, if is value use insert
+  // -check current.next.value
+  // -when we find value insert the new node (temp is current.next, current.next is new node, new node.next is temp)
+  //- return the list
+
+  insertBefore(value, newValue) {
+    var newNode = new Node(newValue);
+    // let current = this.head;
+    if (this.head != value) {
+      while (current) {
+        current = current.next;
+        if (current.next.value === value) {
```

Comment:
When you reach the last node, you are advancing to `null` then trying to read the value of next, which `null` does not have.